### PR TITLE
Move diagnostics package out of bootstrap module

### PR DESCRIPTION
**Pull request by trask**
Bonus: also upgrades to SLF4J 2.x

This will also allow follow-up PR to remove moshi dependency

### CODE REVIEW

File: agent/agent-tooling/build.gradle.kts [hunk 1]
```diff
@@ -25,6 +25,17 @@ dependencies {
     exclude("org.ow2.asm", "asm")
   }
 
+  implementation("ch.qos.logback:logback-classic")
+  implementation("ch.qos.logback.contrib:logback-json-classic")
+
+  // not using gson because it has dependency on java.sql.*, which is not available in Java 9+ bootstrap class loader
+  // only complaint so far about moshi is that it doesn't give line numbers when there are json formatting errors
+  implementation("com.squareup.moshi:moshi")
```
Comment (trask) [inline agent/agent-tooling/build.gradle.kts:33]: this was what motivated moving this code out of the bootstrap module (in addition to it being a generally good thing to minimize amount of classes in the bootstrap module)

File: agent/agent-bootstrap/build.gradle.kts [hunk 2]
```diff
@@ -8,24 +8,4 @@ dependencies {
...
-
-  compileOnly("com.google.auto.value:auto-value-annotations")
-  annotationProcessor("com.google.auto.value:auto-value")
-
-  implementation("ch.qos.logback:logback-classic")
-  implementation("ch.qos.logback.contrib:logback-json-classic")
-
-  // not using gson because it has dependency on java.sql.*, which is not available in Java 9+ bootstrap class loader
-  // only complaint so far about moshi is that it doesn"t give line numbers when there are json formatting errors
-  implementation("com.squareup.moshi:moshi")
```
Comment (trask) [inline agent/agent-bootstrap/build.gradle.kts:21]: this was what motivated moving this code out of the bootstrap module (in addition to it being a generally good thing to minimize amount of classes in the bootstrap module)

(although this comment is now very outdated, and we'll move to jackson)

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/FirstEntryPoint.java [hunk 3]
```diff
@@ -160,6 +161,10 @@ private static String findSystemProperties() {
   }
 
   @Override
+  @SuppressFBWarnings(
+      value = "SECCRLFLOG", // CRLF injection into log messages
+      justification =
+          "Logging params cannot be controlled by an end user of the instrumented application")
```
Comment (trask) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/FirstEntryPoint.java:167]: no idea why this warning only started showing up with this change (https://find-sec-bugs.github.io/bugs.htm#CRLF_INJECTION_LOGS)

File: buildSrc/src/main/kotlin/ai.smoke-test.gradle.kts [hunk 4]
```diff
@@ -74,6 +74,16 @@ dependencies {
...
 }
 
+configurations.all {
+  // spring boot 2.x requires slf4j 1.x
+  val slf4jVersion = "1.7.36"
+  resolutionStrategy.force("org.slf4j:slf4j-api:${slf4jVersion}")
+  resolutionStrategy.force("org.slf4j:log4j-over-slf4j:${slf4jVersion}")
+  resolutionStrategy.force("org.slf4j:jcl-over-slf4j:${slf4jVersion}")
+  resolutionStrategy.force("org.slf4j:jul-to-slf4j:${slf4jVersion}")
+  resolutionStrategy.force("ch.qos.logback:logback-classic:1.2.12")
```
Comment (trask) [inline buildSrc/src/main/kotlin/ai.smoke-test.gradle.kts:84]: need to override dependencyManagement for slf4j version in smoke tests

File: dependencyManagement/build.gradle.kts [hunk 5]
```diff
@@ -40,12 +40,7 @@ val errorProneVersion = "2.19.1"
 val byteBuddyVersion = "1.12.18"
 val jmhVersion = "1.36"
 val mockitoVersion = "4.11.0"
-// moving to 2.0 is problematic because the SPI mechanism in 2.0 doesn't work in the
-// bootstrap class loader because, while we add the agent jar to the bootstrap class loader
-// via Instrumentation.appendToBootstrapClassLoaderSearch(), there's nothing similar for
-// resources (which is a known problem in the java agent world), and so the META-INF/services
-// resource is not found
-val slf4jVersion = "1.7.36"
+val slf4jVersion = "2.0.7"
```
Comment (trask) [inline dependencyManagement/build.gradle.kts:43]: 🥳

File: agent/agent/build.gradle.kts [hunk 6]
```diff
@@ -104,20 +104,20 @@ tasks {
...
-    // (which come from the upstream agent)
-    exclude("inst/io/opentelemetry/javaagent/slf4j/**")
+    dependsOn(isolateJavaagentLibs)
+    from(isolateJavaagentLibs.get().outputs)
+
+    // this distro uses logback, so need to exclude slf4j-simple
+    exclude("inst/io/opentelemetry/javaagent/slf4j/simple/**")
+    // unfortunately, this also excludes the same from our distro (which points to logback)
+    // and so we have to hackily re-add it via agent/agent/src/main/resources
     exclude("inst/META-INF/services/io.opentelemetry.javaagent.slf4j.spi.SLF4JServiceProvider")
```
Comment (trask) [inline agent/agent/build.gradle.kts:114]: yuck 🙈

**Review by jeanbisutti — APPROVED** [review]

**Review by heyams — APPROVED** [review]

**Pull request merged**